### DISCUSSION TITLE
A Problem about using subscribeWithSelector with Vanilla store

### DISCUSSION THREAD

**Discussion by ywleeee** [pmndrs/zustand#1019] [Q&A] [answered]
Hi everyone, I'm new to zustand, and I encounter some problems

I'm using the latest version of Zustand, and as the docs suggest, I can do the following things:
`import { subscribeWithSelector } from 'zustand/middleware' `
`import { createStore, useStore } from "zustand";`
`const store = createStore( subscribeWithSelector (()=>{...myStoreData...}))`

It works, but after I wrap my data in the middleware, I found that
`useStore(store)`
no longer return an object(it returns a function), and I can't write
`useStore(store, selector)` anymore
Could anyone gives me a hint how to fix it? Since I can't find anything about it in the docs
Thanks in advance!

**Comment by dai-shi** [comment, marked as answer]
https://codesandbox.io/s/amazing-wind-2i7bbw?file=/src/App.js
It looks working. Check it out.

**Reply by ywleeee** [reply]
Thanks for the quick reply
Problem is solved
Very appreciate
